ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/SnapProtocol.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth;
+
+import org.hyperledger.besu.ethereum.eth.messages.SnapV1;
+import org.hyperledger.besu.ethereum.p2p.rlpx.wire.Capability;
+import org.hyperledger.besu.ethereum.p2p.rlpx.wire.SubProtocol;
+
+import java.util.List;
+
+/**
+ * Snap protocol messages as defined in https://github.com/ethereum/devp2p/blob/master/caps/snap.md}
+ */
+public class SnapProtocol implements SubProtocol {
+  public static final String NAME = "snap";
+  public static final Capability SNAP1 = Capability.create(NAME, SnapVersion.V1);
+
+  private static final SnapProtocol INSTANCE = new SnapProtocol();
+
+  private static final List<Integer> snap1Messages =
+      List.of(SnapV1.GET_ACCOUNT_RANGE, SnapV1.ACCOUNT_RANGE);
```

Comment:
👍  .  Are these all of the snap protocol messages?

Reply:
No it's just an initilalization PR I just added the first messages in order to test the communication mechanisms and that it make an example for adding the other messages